gamemodes/terrortown/entities/weapons/weapon_zm_carry.lua

```diff
@@ -222,11 +205,9 @@ SWEP.reset = SWEP.Reset
 -- @return boolean
 -- @realm shared
 function SWEP:CheckValidity()
-	if not IsValid(self.EntHolding) or not IsValid(self.CarryHack) or not IsValid(self.Constr) then
-		-- if one of them is not valid but another is non-nil...
-		if self.EntHolding or self.CarryHack or self.Constr then
-			self:Reset()
-		end
+	local ctarget = self:GetCarryTarget()
+	if ctarget ~= NULL and not IsValid(ctarget) then
```

Comment:
```suggestion
	if ctarget and not IsValid(ctarget) then
```

This basically checks if an entity exists that is an null-entity. That is kind of weird to me

Reply:
it can only ever be null, even if it's set as nil on the server but this is fine